Sources/SwiftInspectorVisitors/Tests/TypeSyntax+NameExtractorSpec.swift

```diff
@@ -0,0 +1,111 @@
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+import Foundation
+import Nimble
+import Quick
+import SwiftInspectorTestHelpers
+import SwiftSyntax
+
+@testable import SwiftInspectorVisitors
+
+final class TypeSyntaxNameExtractorSpec: QuickSpec {
+  private var sut = EnumVisitor()
+
+  override func spec() {
+    beforeEach {
+      self.sut = EnumVisitor()
+    }
+
+    describe("qualifiedName") {
+      context("when called on a TypeSyntax node representing a SimpleTypeIdentifierSyntax") {
+        final class SimpleTypeIdentifierSyntaxVisitor: SyntaxVisitor {
+          var simpleTypeIdentifier: String?
+          override func visit(_ node: SimpleTypeIdentifierSyntax) -> SyntaxVisitorContinueKind {
+            simpleTypeIdentifier = TypeSyntax(node).qualifiedName
+            return .skipChildren
+          }
+        }
+
+        var visitor: SimpleTypeIdentifierSyntaxVisitor!
+        beforeEach {
+          let content = """
+              var int: Int = 1
+              """
+
+          visitor = SimpleTypeIdentifierSyntaxVisitor()
+          try? VisitorExecutor.walkVisitor(visitor, overContent: content)
+        }
+
+        it("Finds the type") {
+          expect(visitor?.simpleTypeIdentifier) == "Int"
+        }
+      }
+
+      context("when called on a TypeSyntax node representing a MemberTypeIdentifierSyntax") {
+        final class MemberTypeIdentifierSyntaxVisitor: SyntaxVisitor {
+          var memberTypeIdentifier: String?
+          override func visit(_ node: MemberTypeIdentifierSyntax) -> SyntaxVisitorContinueKind {
+            memberTypeIdentifier = TypeSyntax(node).qualifiedName
+            return .skipChildren
+          }
+        }
+
+        var visitor: MemberTypeIdentifierSyntaxVisitor!
+        beforeEach {
+          let content = """
+              var int: Swift.Int = 1
+              """
+
+          visitor = MemberTypeIdentifierSyntaxVisitor()
+          try? VisitorExecutor.walkVisitor(visitor, overContent: content)
+        }
+
+        it("Finds the type") {
+          expect(visitor?.memberTypeIdentifier) == "Swift.Int"
+        }
+      }
+
+      context("when called on a TypeSyntax node representing an Array") {
+        final class ArraySyntaxVisitor: SyntaxVisitor {
+          override func visit(_ node: ArrayTypeSyntax) -> SyntaxVisitorContinueKind {
+            // This line should assert.
+            _ = TypeSyntax(node).qualifiedName
+            fail("This line should never be reached")
+            return .skipChildren
+          }
+        }
+
+        it("asserts") {
+          expect(try VisitorExecutor.walkVisitor(
+                  ArraySyntaxVisitor(),
+                  overContent: """
+                    var intArray: [Int] = [Int]()
```

Comment:
One random thought... Functionally `[Int]` is equivalent to `Array<Int>`. I wonder if we need to handle this case when we think about properties. I would expect that most users of SwiftInspector may not care if the code says `[Int]` or `Array<Int>`.

Reply:
Indeed. The good news is that our current code doesn't yet inspect an `ArrayTypeSyntax`, and the `qualifiedName` property is `internal` so if we end up tripping the assertion it's because we've foot-gunned.

But yes, there's room to make this code more compile-time safe.

Reply:
And in addition to compile-time safe I am hoping we can make the output as easy as possible to consume. In most cases I suspect you won't care if a type is `[Int]` or `Array<Int>`.